core/wasm-interface/src/lib.rs

```diff
@@ -0,0 +1,285 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Types and traits for interfacing between the host and the wasm runtime.
+
+use std::{borrow::Cow, marker::PhantomData, mem, iter::Iterator};
+
+mod wasmi_impl;
+
+/// Value types supported by Substrate on the boundary between host/Wasm.
+#[derive(Copy, Clone, PartialEq, Debug, Eq)]
+pub enum ValueType {
+	/// An `i32` value type.
+	I32,
+	/// An `i64` value type.
+	I64,
```

Comment:
I assume that it was actually decided that we do not longer want to support host functions that return or receive float values.